cEP-0024.md

```diff
@@ -0,0 +1,215 @@
+| Version  | 1.0                                                   |
+| Title    | Implement Gitmate automations and plugins for coala   |
+| Authors  | Vamshi Krishna Bommerla <mailto:bommvams72@gmail.com> |
+| Status   | Proposed                                              |
+| Type     | Process                                               |
+
+## Abstract
+
+This cEP describes the improvements to the coala workflow and the necessary
+automation in [GitMate](https://gitmate.io/) to realise it, as a part of the
+[GSoC project](https://summerofcode.withgoogle.com/projects/#5344316227059712).
+
+## Introduction
+
+[GitMate](https://gitmate.io/) is an automation tool for developers. While it
+works well, there are a number of missing features which would save time of
+coala community, for example automatically close PRs opened on master of forks.
+Many of the plugins in GitMate are general plugins, which are required by every
+organization, but coala needs more plugins which support coala development
+workflow. This project adds plugins to auto-review, auto-reject, auto-label
+PRs, assign, unassign issues, squash commits.
+
+## New Features
+
+### Auto-assign issue when it is mentioned in any of PR commits
+
+When a PR is opened(or updated), if any of PR's commit message referenced
+any issues with keywords `Fixes` or `Closes` and there is no
+CI failure in PR, then auto assign PR author to the mentioned unassigned
+issues.
+
+If someone else is working on the mentioned issue, comment will be added
+on the PR (configurable).
+
+If PR author is not a member of the organization, then he/she can't be assigned
+to any of the issues, a label(configurable) will be added on the PR so that
+maintainers can identify these type of PRs and a comment will be added on PR
+informing PR author to get invited to the organization before working on any
+issue.
+
+coala can configure this plugin as:
+
+```
+Comment message when someone else is assigned to mentioned issues(use
+`{issues}` as placeholder to get links of all assigned issues):
+Sorry, but the issues {issues} are assigned to someone else. Please try working
+on some other issues.
+
+Label to represent that PR author is not a member of the org: author/non-member
+```
+
+### Auto-reject PR opened on certain branch names
+
+Normally, contributors should make changes in different branch from master
+branch of the forked repository, and submit a PR. But, often coala receives PRs
+from master branch of the forked repository.
+
+coala needs mechanism such that when a PR is opened or updated, GitMate
+automatically closes PRs opened on certain branch names(master), adds certain
+label to the PR so that maintainers can identify these type of PRs and comments
+on the PR informing the PR author to work on a different branch. Comment
+message, branch names, label will be configurable.
+
+coala can configure this plugin as:
+
+```
+Branch names: master
```

Comment:
split into three separate configuration chucks, and only use backticks for literal values.

dont put configuration names (keys) into literal code blocks.  Use bold, or something similar, to show they are customisation field labels.